src/routes/(app)/read-notifications/+server.ts

```diff
@@ -0,0 +1,45 @@
+import apiNames from "$lib/utils/apiNames";
+import { isAuthorized } from "$lib/utils/authorization";
+import type { RequestHandler } from "./$types";
+
+// Read notification
+const readNotifications: RequestHandler = async ({ locals, url }) => {
```

Comment:
So I re-built how "read notifications" is done. From an abstract perspective, marking notifications as read is not an explicit user actions, it is done when they click on the bell icon. Previously, this bell icon was within a `<form>` tag, and when clicked, a form would be submitted.

This worked because your browser has javascript enabled, and what superforms (or SvelteKit for that matter) does when javascript is enabled, is that they stop the form submission and instead just sends a fetch request to the server. Regular form requests always lead to a page load. This means notifications would not work for people without JS enabled. Even if those users are few, just semantically this did not make to be a form.

Instead I built a `PUT` endpoint which is called when the bell icon is clicked, marking all notifications as read.

Reply:
Of course, this means non-JS users can not mark their notifications as read. That's a bummer and if necessary, we might want to add some support for that in the future (like a "mark as read" button which is actually a form)

Reply:
Good point!